server/controllers/crimes.js

```diff
@@ -0,0 +1,68 @@
+const Crimes = require('../models/crimes');
+
+const getAllCrimes = async (ctx) => {
+  try {
+    const res = await Crimes.find({}); 
+    ctx.body = res;
+    ctx.status = 200;
+  } catch (e) {
+    ctx.status = 500;
+  }
+
+};
+
+const getCrimes = async (ctx) => {
+  try {
+
+    const coords = JSON.parse(ctx.params.coords);
+    const res = await Crimes.find({
+      location: {
+        $geoWithin: {
```

Comment:
This controllers nows a lot about the data layer: it's too coupled. It knows the DB is a MongoDB instance and it even uses a very specific operator. What happens if we decide to change the data layer? You have to change the business logic (controllers) layer too, introducing a lot of bugs for sure.

It's better to create a method in the `Crimes` model: `Crimes.findByLocation(coords)`.